crates/forge/tests/cli/test_cmd.rs

```diff
@@ -3649,3 +3650,75 @@ Encountered a total of 1 failing tests, 0 tests succeeded
 
 "#]]);
 });
+
+forgetest!(test_eip712_cheatcode, |prj, cmd| {
+    prj.add_source(
+        "Eip712",
+        r#"
+contract Eip712Structs {
+    struct Transaction {
+        Person from;
+        Person to;
+        Asset tx;
+    }
+    struct Person {
+        address wallet;
+        string name;
+    }
+    struct Asset {
+        address token;
+        uint256 amount;
+    }
+}
+    "#,
+    )
+    .unwrap();
+    prj.insert_ds_test();
+    prj.insert_vm();
+    prj.insert_console();
+
+    prj.add_source("Eip712Cheat.sol", r#"
+// Note Used in forge-cli tests to assert failures.
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity ^0.8.18;
+
+import "./test.sol";
+import "./Vm.sol";
+import "./console.sol";
+
+string constant CANONICAL = "Transaction(Person from,Person to,Asset tx)Asset(address token,uint256 amount)Person(address wallet,string name)";
+
+contract Eip712Test is DSTest {
+    Vm constant vm = Vm(HEVM_ADDRESS);
+
+    function testEip712HashType() public {
+        bytes32 canonicalHash = keccak256(bytes(CANONICAL));
+
+        // Can figure out the canonical type from a messy string representation of the type,
```

Comment:
Would you mind adding a simple example without nested structs as a smoke test?

e.g.

`"EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)"`

and accompany it with a comment displaying the result

```
chisel
➜ keccak256("EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)")
Type: bytes32
└ Data: 0x8b73c3c69bb8fe3d512ecc4cf759cc79239f7b179b0ffacaa9a75d522b39400f
```

ref:

https://github.com/Vectorized/solady/blob/b609a9c79ce541c2beca7a7d247665e7c93942a3/src/utils/EIP712.sol#L20-L22